docs/cmd/kn_service_update.md

```diff
@@ -38,28 +38,29 @@ kn service update NAME [flags]
-      --traffic strings          Set traffic distribution (format: --traffic revisionRef=percent) where revisionRef can be a revision or a tag or '@latest' string representing latest ready revision. This flag can be given multiple times with percent summing up to 100%.
-      --untag strings            Untag revision (format: --untag tagName). This flag can be spcified multiple times.
-      --wait-timeout int         Seconds to wait before giving up on waiting for service to be ready. (default 60)
+      --async                         Update service and don't wait for it to become ready.
+      --concurrency-limit int         Hard Limit of concurrent requests to be processed by a single replica.
+      --concurrency-target int        Recommendation for when to scale up based on the concurrent number of incoming request. Defaults to --concurrency-limit when given.
+  -e, --env stringArray               Environment variable to set. NAME=value; you may provide this flag any number of times to set multiple environment variables. To unset, specify the environment variable name followed by a "-" (e.g., NAME-).
+  -h, --help                          help for update
+      --image string                  Image to run.
+  -l, --label stringArray             Service label to set. name=value; you may provide this flag any number of times to set multiple labels. To unset, specify the label name followed by a "-" (e.g., name-).
+      --limits-cpu string             The limits on the requested CPU (e.g., 1000m).
+      --limits-memory string          The limits on the requested memory (e.g., 1024Mi).
+      --lock-to-digest                keep the running image for the service constant when not explicitly specifying the image. (--no-lock-to-digest pulls the image tag afresh with each new revision) (default true)
+      --max-scale int                 Maximal number of replicas.
+      --min-scale int                 Minimal number of replicas.
+  -n, --namespace string              List the requested object(s) in given namespace.
+      --no-lock-to-digest             do not keep the running image for the service constant when not explicitly specifying the image. (--no-lock-to-digest pulls the image tag afresh with each new revision)
+  -p, --port int32                    The port where application listens on.
+      --requests-cpu string           The requested CPU (e.g., 250m).
+      --requests-memory string        The requested memory (e.g., 64Mi).
+      --revision-name string          The revision name to set. Must start with the service name and a dash as a prefix. Empty revision name will result in the server generating a name for the revision. Accepts golang templates, allowing {{.Service}} for the service name, {{.Generation}} for the generation, and {{.Random [n]}} for n random consonants. (default "{{.Service}}-{{.Random 5}}-{{.Generation}}")
+      --service-account-name string   Service account name. To unset, specify "-". (default "-")
```

Comment:
Does this mean, that when you use `kn service update` without this option, that the service account will be always removed because the default is `-` ? I would expect that there is no default and service account name is left untouched when this option is not given.

Reply:
For `kn service update` I would prefer:

* `--service-account new-name` : Update the service account with the new name
* `--service-account` : Clear the service account
* Option not given : Don't touch service account

Reply:
@rhuss I've made a change according to your comments. Please check it out and, if there are additional comments, please let me know.